### PR TITLE
[foreman] Collect debugging info for Dynflow Sidekiq

### PR DESCRIPTION
**Pull request by pmoravec**
Collect sidekiq logs and services lists and statuses.

Resolves: #1883

Signed-off-by: Pavel Moravec <pmoravec@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [X] If this commit closes an existing issue, is the line `Closes: #ISSUENUMBER` included in an independent line?
- [X] If this commit resolves an existing pull request, is the line `Resolves: #PRNUMBER` included in an independent line?

### CODE REVIEW

**Comment by pmoravec** [comment]
This is in par with `foreman-debug` stuff in https://github.com/theforeman/foreman/pull/7192 (`/etc/foreman/dynflow` is already collected via collecting `/etcforeman`).

**Comment by pmoravec** [comment]
The `"system-dynflow\\x2dsidekiq.slice"` is really required, as the `systemd` unit name contains the "\x2d", not the "-".

File: sos/plugins/foreman.py [hunk 1]
```diff
@@ -139,6 +139,14 @@ def setup(self):
             'ping -c1 -W1 localhost'
         ])
 
+        # Dynflow Sidekiq
+        self.add_cmd_output('systemctl list-units dynflow*',
+                            suggest_filename='dynflow_units')
+        self.add_cmd_output('systemctl status '
+                            '"system-dynflow\\x2dsidekiq.slice"',
+                            suggest_filename='dynflow_sidekiq_status')
```
Comment (TurboTurtle) [inline sos/plugins/foreman.py]: The status request can be handled by the `Plugin.add_service_status()` method that hooks into the `InitSystem` abstraction. Though it does look like we'll need to plumb `**kwargs` through so `suggest_filename` can be passed properly.

**Comment by pmoravec** [comment]
Good point, `add_service_status` now supports all `kwargs`. That causes a slight difference in behaviour: originally `add_service_status` called the command with `timeout=None`, now it calls with the default timeout. But I see this as an improvement rather than a regression. (removal of `pred=None` does not have an effect since that is the predicate default already).

**Comment by pmoravec** [comment]
Just tested on upstream foreman, works fine.

**Pull request closed without merging**